### DISCUSSION TITLE
Can you show an example to how to set a value to an element

### DISCUSSION THREAD

**Discussion by L150n** [vueform/vueform#170] [Questions] [answered]
I would like to know how to set values for a text element rather than using the default value. For example:

` <Vueform>
    <TextElement
      name="text"
      label="Text"
 />
  </Vueform>`
  
  In this scenario, I want to set a value for the text element when an action or a button is pressed.

**Comment by adamberecz** [comment, marked as answer]
There are multiple ways:

## Using ref

https://vueform.com/docs/handling-form-data#managing-form-data-via-api

```vue
<template>
   <Vueform ref="form$">
    <TextElement name="text" label="Text" />
  </Vueform>
</template>

<script setup>
import { ref } from 'vue'

const form$ = ref(null)

onMounted(() => {
  form$.value.el$('text').update('x')
  // or
  form$.value.update({
    text: 'x'
  })
}) 
</script>
```

## Using v-model with sync

https://vueform.com/docs/handling-form-data#synchronizing-external-data


```vue
<template>
   <Vueform v-model="data" sync>
    <TextElement name="text" label="Text" />
  </Vueform>
</template>

<script setup>
import { ref } from 'vue'

const data = ref({})

onMounted(() => {
  data.value.text = 'x'
}) 
</script>
```
